playbooks/roles/filebeat/tasks/main.yml

```diff
@@ -0,0 +1,53 @@
+---
+
+- name: Install apt package dependencies.
+  apt:
+    name: apt-transport-https
+    state: present
+
+- name: Add Elastic's apt key
+  apt_key:
+    url: https://artifacts.elastic.co/GPG-KEY-elasticsearch
+    state: present
+
+- name: Add Elastic's apt repository
+  apt_repository:
+    repo: 'deb https://artifacts.elastic.co/packages/6.x/apt stable main'
+    state: present
+    update_cache: yes
+
+- name: Install Filebeat
+  apt:
+    name: filebeat
+    state: present
+
+- name: Install Filebeat configuration
+  template:
+    src: filebeat.yml.j2
+    dest: "{{ filebeat_config_dir }}/filebeat.yml"
+  notify:
+    - restart filebeat
+
+- name: Install Filebeat CA certificate
+  copy:
+    content: "{{ filebeat_ca_cert }}"
+    dest: "{{ filebeat_ca_path }}"
+    owner: root
+    group: root
+    mode: 0400
+  when: filebeat_ca_cert != "" and filebeat_ca_cert_url == ""
+
+- name: Install Filebeat CA certificate from URL
+  get_url:
+    url: "{{ filebeat_ca_cert_url }}"
+    dest: "{{ filebeat_ca_path }}"
+    owner: root
+    group: root
+    mode: 0400
+  when: filebeat_ca_cert == "" and filebeat_ca_cert_url != ""
```

Comment:
So if someone sets both `filebeat_ca_cert` and `filebeat_ca_cet_url`, none of them will be used?  Maybe add an `assert` to give an error message in that case – silently doing nothing instead seems somewhat unexpected.

Reply:
Yeah didn't think about that. Will add an assert.